content/3_Silver/Graph_Traversal.mdx

```diff
@@ -1082,6 +1082,47 @@ public class BuildingTeams {
 ```
 
 </JavaSection>
+<PySection>
+
+Note that the below solution passes in CPython3 but exceeds the time limit on test case #7 in PyPy3.
+
+```py
+import sys
+
+input = sys.stdin.readline
+
+sys.setrecursionlimit(int(1e5 + 1))
+
+n, m = map(int, input().strip().split())
+adj = [[] for _ in range(n)]
+team = [0] * n  # 0 -> has not been assigned a team yet
+
+for _ in range(m):
+	a, b = map(int, map(int, input().strip().split()))
+	adj[a - 1].append(b - 1)
+	adj[b - 1].append(a - 1)
+
+
+def dfs(node):
+	for next_node in adj[node]:
+		if team[next_node]:
+			if team[next_node] == team[node]:
+				print("IMPOSSIBLE")
+				raise SystemExit
```

Comment:
u can also use exit(0)

Reply:
Yep, thanks. I like to use SystemExit just due to preference lol.

Although I do think it is slightly faster and more descriptive. https://stackoverflow.com/a/35430500

Reply:
wait this is the first time i've seen this used what

Reply:
ok so for consistency with rest of code could you use exit()